### PR TITLE
Change nft owner and spender identifier from long to AccountID

### LINKED ISSUE #7244: Update nft.proto and classes that rely on it to use `AccountID` and `TokenID` instead of long

### PR DESCRIPTION
**Pull request by kimbor**
**Description**:
Use IDs instead of longs in NFT

**Related issue(s)**:

Fixes #7244 
Related to [PR #289](https://github.com/hashgraph/hedera-protobufs/pull/289) in hedera-protobufs repo 

**Notes for reviewer**:

This PR relies on a temporary branch `change-numbers-to-ids-nft-proto ` in hedera-protobufs repo. The plan is to:
1) Merge this PR
2) Merge [PR #289](https://github.com/hashgraph/hedera-protobufs/pull/289) in hedera-protobufs repo, but DO NOT DELETE THE ASSOCIATED BRANCH YET. Until Step 3 is completed, hedera-services still relies on the branch
3) Create and merge another PR in hedera-services to reset the protobufs branch in use back to add-pbj-types-for-state
4) Delete the hedera-protobufs branch associated with step 2

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java [hunk 1]
```diff
@@ -396,10 +396,15 @@ public void updateSpender(
...
             final var token = tokenStore.get(tokenId);
 
-            validateTrue(isValidOwner(nft, owner.accountNumber(), token), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
-            final var copy = nft.copyBuilder()
-                    .spenderNumber(spenderId.accountNumOrThrow())
+            // FUTURE: owner will be updated to use an AccountID instead of a long account number
+            // Issue #7243
+            AccountID accountOwner = AccountID.newBuilder()
+                    .accountNum(owner.accountNumber())
+                    .realmNum(0)
```
Comment (jsync-swirlds) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java:403]: Suggestion:
Pull shard and realm from the payer account ID in the TransactionID, so we aren't hard-coding a value that has to be changeable.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java [hunk 2]
```diff
@@ -152,10 +153,18 @@ private void deleteNftSerials(
 
                 validateTrue(nft != null, INVALID_NFT_ID);
                 // Check if owner owns the nft
-                validateTrue(isValidOwner(nft, owner.accountNumber(), token), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
+                // FUTURE: owner will be updated to use an AccountID instead of a long account number
+                // Issue #7243
+                AccountID accountOwner = AccountID.newBuilder()
```
Comment (jsync-swirlds) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java:158]: We already have owner (Account) from line 149, couldn't we use the AccountID from that?
Also, if we cannot use the owner, we should get the shard and realm from the tokenId on the assumption the shard/realm of the account matches the token it owns.

Comment (kimbor) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java:158]: The plan is to use the owner Account from line 149. Currently Account does not have a getAccountId() method, only a getAccountNum(). Issue #7243 will correct that. The code here is very temporary until issue #7243 can be implemented, probably this week.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java (hunk 1 above)
Comment (kimbor) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java:403]: Owner Account does not have the ID yet. That's what issue #7243 is supposed to fix. When #7243 is implemented, this code should be updated to use that. #7243 is probably the next issue I will work on after this, so this hack of putting 0 in for realm and shard is very temporary.

**Review by mhess-swl — APPROVED** [review]

**Review by jsync-swirlds — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**